### PR TITLE
create resource search

### PR DESCRIPTION
**Pull request by ThiagoLopes**
Implementation new view, url and template

### CODE REVIEW

File: inspace/frontend/templates/core/home.html [hunk 1]
```diff
@@ -6,10 +6,10 @@
     <div class="columns">
       <div class="column is-half is-offset-one-quarter">
         <h1 class="title is-1">InSpace</h1>
-        <form class="control" action="results.html">
+        <form class="control" method="GET" action="{% url 'core:resource-list' title%}">
```
Comment (cacarrara) [inline inspace/frontend/templates/core/home.html]: Why this final "title" in action url?

File: inspace/frontend/templates/core/resource_list.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{% extends 'base.html' %} {% block base_content %}
```
Comment (cacarrara) [inline inspace/frontend/templates/core/resource_list.html]: Left one blank line between extends and next statement.

File: inspace/frontend/templates/core/resource_list.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      {% for resource in resource_list %}
+      <li>
+        {{resource}}
+      </li>
+      {% endfor %}
+    </ul>
+
+  </div>
+</section>
+{%endblock%}
```
Comment (cacarrara) [inline inspace/frontend/templates/core/resource_list.html]: Spaces before and after `endblock` keyword.

File: inspace/frontend/templates/core/resource_list.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+<section class="section hero has-text-centered">
+  <div class="hero-body">
+    <h1 class="title">
+      {{request.method}}
+    </h1>
+
+    <ul>
+      {% for resource in resource_list %}
+      <li>
+        {{resource}}
```
Comment (cacarrara) [inline inspace/frontend/templates/core/resource_list.html]: Spaces before and after `resource`

File: inspace/tests/core/test_views.py [hunk 5]
```diff
@@ -1,6 +1,14 @@
...
 def test_home_get_successfully(client):
     response = client.get(reverse('core:home'))
     assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_resource_list_successfully(client):
+    response = client.get(
+        reverse('core:resource-list', kwargs={'title': 'python'}))
+    assert response.status_code == 200
```
Comment (cacarrara) [inline inspace/tests/core/test_views.py]: More tests, please. As I asked in #1

File: inspace/core/views.py [hunk 6]
```diff
@@ -18,5 +18,18 @@ class ResourceCreateView(CreateView):
     form_class = ResourceForm
     context_object_name = 'reosource'
 
-
```
Comment (cacarrara) [inline inspace/core/views.py]: We need to keep this line (pep8).

File: inspace/core/views.py [hunk 7]
```diff
@@ -18,5 +18,18 @@ class ResourceCreateView(CreateView):
...
+    http_method_names = ('get', )
+    model = Resource
+    template_name = 'core/resource_list.html'
+    context_object_name = 'resource_list'
+
+    def get_queryset(self):
+        title = self.request.GET.get('title')
+        # QUERY TABELA
+        return self.model.objects.all()
+
```
Comment (cacarrara) [inline inspace/core/views.py]: We need to keep 2 lines after function and class definitions (pep8).

File: inspace/core/views.py [hunk 8]
```diff
@@ -18,5 +18,18 @@ class ResourceCreateView(CreateView):
...
 
-
 resource_create_view = ResourceCreateView.as_view()
+
+
+class ResourceListView(ListView):
+    http_method_names = ('get', )
+    model = Resource
+    template_name = 'core/resource_list.html'
+    context_object_name = 'resource_list'
```
Comment (cacarrara) [inline inspace/core/views.py]: I think resource_list is the default object name in context, so you do not need to define it here.

File: inspace/core/views.py [hunk 9]
```diff
@@ -18,5 +18,18 @@ class ResourceCreateView(CreateView):
...
     context_object_name = 'reosource'
 
-
 resource_create_view = ResourceCreateView.as_view()
+
+
+class ResourceListView(ListView):
+    http_method_names = ('get', )
+    model = Resource
+    template_name = 'core/resource_list.html'
```
Comment (cacarrara) [inline inspace/core/views.py]: Did you try do not specify template name here? I think the `app_name/model_list.html` is the Django default infered.

File: inspace/core/urls.py [hunk 10]
```diff
@@ -4,6 +4,9 @@
 
 
 urlpatterns = [
-    url(r'^resources/new/$', views.resource_create_view, name='resource-create'),
+    url(r'^resources/new/$',
+        views.resource_create_view, name='resource-create'),
+    url(r'^resources/(?P<title>\w{0,50})$',
```
Comment (ThiagoLopes) [inline inspace/core/urls.py]: Sorry, my bad. it is not necessary

File: inspace/core/views.py (hunk 9 above)
Comment (ThiagoLopes) [inline inspace/core/views.py]: Yeap!

File: inspace/core/views.py (hunk 8 above)
Comment (ThiagoLopes) [inline inspace/core/views.py]: My bad, [explanation](https://docs.djangoproject.com/en/1.11/ref/class-based-views/mixins-multiple-object/#django.views.generic.list.MultipleObjectMixin.get_context_object_name)

File: inspace/core/urls.py (hunk 10 above)
Comment (cacarrara) [inline inspace/core/urls.py]: Hehe, you do not need to sorry dude. Relax.

File: inspace/core/views.py (hunk 8 above)
Comment (cacarrara) [inline inspace/core/views.py]: Yep :smile_cat: we do not need it.

File: inspace/core/views.py [hunk 11]
```diff
@@ -1,5 +1,5 @@
-from django.views.generic import CreateView, TemplateView
-
+from django.views.generic import CreateView, TemplateView, ListView
+from django.shortcuts import redirect
```
Comment (cacarrara) [inline inspace/core/views.py]: You're not using redirect.

File: inspace/tests/core/test_views.py (hunk 5 above)
Comment (ThiagoLopes) [inline inspace/tests/core/test_views.py]: to make this we need create [this](https://code.djangoproject.com/wiki/Fixtures)?

File: inspace/tests/conftest.py [hunk 12]
```diff
@@ -1,7 +1,16 @@
...
+
+from core.models import Resource
 
 
 @pytest.fixture
 def client():
     return Client()
+
+@pytest.fixture
+def resources():
```
Comment (cacarrara) [inline inspace/tests/conftest.py]: Feature is called `resources` and you return only one `Resource`. You should return the resources list here.

File: inspace/tests/core/test_views.py [hunk 13]
```diff
@@ -1,6 +1,31 @@
...
 
 
 def test_home_get_successfully(client):
     response = client.get(reverse('core:home'))
     assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_resource_list_successfully(client):
+    """test 200 in resource_list"""
```
Comment (cacarrara) [inline inspace/tests/core/test_views.py]: This is not a necessary comment.

File: inspace/tests/core/test_views.py [hunk 14]
```diff
@@ -1,6 +1,31 @@
...
+    response = client.get(
+        reverse('core:resource-list'))
+    assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_resources_number_all(client, resources):
+    """return all resources"""
```
Comment (cacarrara) [inline inspace/tests/core/test_views.py]: Comment is not necessary.

File: inspace/tests/core/test_views.py [hunk 15]
```diff
@@ -1,6 +1,31 @@
...
+    response = client.get(
+        reverse('core:resource-list'))
+    # will return all resources
```
Comment (cacarrara) [inline inspace/tests/core/test_views.py]: Unecessary comment here too.

File: inspace/tests/core/test_views.py [hunk 16]
```diff
@@ -1,6 +1,31 @@
...
+    assert len(response.context['resource_list']) == 3
+
+
+@pytest.mark.django_db
+def test_resouces_query_icontains(client, resources):
```
Comment (cacarrara) [inline inspace/tests/core/test_views.py]: Good test :+1:

**Comment by cacarrara** [comment]
I'm approving and merging it, but I also left some important comments.

Thanks !

File: inspace/tests/core/test_views.py (hunk 15 above)
Comment (ThiagoLopes) [inline inspace/tests/core/test_views.py]: oops... i forgot it :smile:

**Pull request merged**